src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -385,23 +385,23 @@ public SubstrateDispatcher(Path buildRoot, ProjectConfiguration config) throws I
 
         Triplet targetTriplet = config.getTargetTriplet();
 
-        this.targetConfiguration = Objects.requireNonNull(getTargetConfiguration(targetTriplet),
+        this.targetConfiguration = Objects.requireNonNull(getTargetConfiguration(targetTriplet, javaVersion),
```

Comment:
Since Java/GraalVM versions are determined (also after #1234) from `InternalProjectConfiguration`, don't you think it should be better to keep the `javaVersion` as a field there, and then add a getter?
That would definitely prevent from needing to pass it all over the place.

In fact, we already have `usesJDK11`. Probably better to remove this one and just have `javaVersion` that can be queried from anywhere.

Reply:
To do that cleanly, I'd have to execute the version parsing logic in the constructor of the `InternalProjectConfiguration`. Would that be acceptable? I'm not sure what the reason is to delay it at the moment.

Reply:
That's fine, so far we do in `SubstrateDispatcher`:

```
        this.config = new InternalProjectConfiguration(config);
        Version javaVersion = this.config.checkGraalVMJavaVersion();
        this.config.checkGraalVMVersion(javaVersion);
        this.config.checkGraalVMVendor();
```

so all of that can go into the constructor, indeed (which already does for instance `performHostChecks`)